src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -633,12 +644,155 @@ public Snapshot makeSnapshotFromSnapshotRequest(SnapshotRequestModel snapshotReq
         .tags(TagUtils.sanitizeTags(snapshotRequestModel.getTags()));
   }
 
-  private void createSnapshotTablesFromDatasetAsset(
-      String datasetName, String assetName, SnapshotSource snapshotSource, Snapshot snapshot) {
-    Dataset queryDataset = datasetService.retrieveByName(datasetName);
-    AssetSpecification queryAssetSpecification = getAssetByNameFromDataset(queryDataset, assetName);
-    snapshotSource.assetSpecification(queryAssetSpecification);
-    conjureSnapshotTablesFromAsset(snapshot, snapshotSource);
+  public SnapshotAccessRequestResponse getSnapshotAccessRequestById(UUID accessRequestId) {
+    return snapshotRequestDao.getById(accessRequestId);
+  }
+
+  public AssetSpecification buildAssetFromSnapshotAccessRequest(
+      Dataset dataset, SnapshotAccessRequestResponse snapshotRequestModel) {
+    // build asset model from snapshot request
+    AssetModel assetModel =
+        new AssetModel()
+            .name("snapshot-by-request-asset")
+            .rootTable("person")
+            .rootColumn("person_id")
+            .followStrictDirection(true);
+    // Manually add dictionary tables, leave columns empty to return all columns
+    assetModel.addTablesItem(new AssetTableModel().name("person"));
+
+    // Build asset tables and columns based on the concept sets included in the snapshot request
+    List<SnapshotBuilderTable> tables = pullTables(snapshotRequestModel);
+    tables.forEach(
+        table -> {
+          assetModel.addTablesItem(
+              new AssetTableModel().name(table.getDatasetTableName()).columns(table.getColumns()));
+          // First add all person <-> occurrence relationships
+          assetModel.addFollowItem(table.getPersonRelationship());
+        });
+    // Second, add all occurrence <-> concept relationships
+    tables.forEach(table -> table.getConceptRelationships().forEach(assetModel::addFollowItem));
```

Comment:
Note for reviewers: With the change to require strict relationships walking, I don't think we _need_ to break this into  person and then concept relationships. However, it might make it easier to correctly define the relationships in the snapshot builder settings for data custodians: It is clear that you need a person -> occurrence relationship and then occurrence -> concept relationship.